File: iphone/Classes/Console.m
//
//  Console.m
//  PhoneGap
//
//  Created by Michael Nachbaur on 14/03/09.
//  Copyright 2009 Decaf Ninja Software. All rights reserved.
//

#import "Console.h"

@implementation Console

+ (void)log:(NSString*)options forWebView:(UIWebView*)webView
{
	NSArray* arguments = [options componentsSeparatedByString:@"/"];
    NSLog(@"[%@] %@", [arguments objectAtIndex:0], [[arguments objectAtIndex:1] stringByReplacingPercentEscapesUsingEncoding:NSUTF8StringEncoding]);
}

+ (void)alert:(NSString*)options forWebView:(UIWebView*)webView
{
	NSArray* arguments = [options componentsSeparatedByString:@"/"];
	NSString* message  = [[arguments objectAtIndex:0] stringByReplacingPercentEscapesUsingEncoding:NSUTF8StringEncoding];
	NSString* title    = @"Alert";
	NSString* button   = @"OK";

	if ([arguments count] >= 1)
		title = [[arguments objectAtIndex:1] stringByReplacingPercentEscapesUsingEncoding:NSUTF8StringEncoding];
	if ([arguments count] >= 2)
		button = [[arguments objectAtIndex:2] stringByReplacingPercentEscapesUsingEncoding:NSUTF8StringEncoding];

	UIAlertView *openURLAlert = [[UIAlertView alloc]
								 initWithTitle:title
								 message:message delegate:nil cancelButtonTitle:button otherButtonTitles:nil];
	[openURLAlert show];
	[openURLAlert release];
}

+ (void)activityStart:(NSString*)options forWebView:(UIWebView*)webView
{
    [(UIActivityIndicatorView*)[webView.window viewWithTag:2] startAnimating];
}

+ (void)activityStop:(NSString*)options forWebView:(UIWebView*)webView
{
    [(UIActivityIndicatorView*)[webView.window viewWithTag:2] stopAnimating];
}

@end

File: javascripts/iphone/console.js

Console.prototype.log = function(message) {
    if (PhoneGap.available)
        PhoneGap.exec('Console.log', 'LOG', this.processMessage(message));
    else
        console.log(message);
};
Console.prototype.error = function(message) {
    if (PhoneGap.available)
        PhoneGap.exec('Console.log', 'ERR', this.processMessage(message));
    else
        console.error(message);
};
Console.prototype.alert = function(message, title, buttonLabel) {
    if (title == undefined || title == null)
        title = 'Alert';
    if (buttonLabel == undefined || buttonLabel == null)
        buttonLabel = 'OK';
    if (PhoneGap.available)
        PhoneGap.exec('Console.alert', message, title, buttonLabel);
    else
        alert(message);
}

Console.prototype.activityStart = function() {
    PhoneGap.exec("Console.activityStart");
};
Console.prototype.activityStop = function() {
    PhoneGap.exec("Console.activityStop");
};
